content/src/denylist/DenylistServiceDecorator.ts

```diff
@@ -282,6 +299,44 @@ export class DenylistServiceDecorator implements MetaverseContentService {
     return types.map((entityType) => buildEntityTarget(entityType, entityId))
   }
 
+  /** Filter out denylisted targets */
+  private async filterDenylisted<T>(
+    denylistRepo: DenylistRepository,
+    elements: T[],
+    targetBuild: (element: T) => DenylistTarget
+  ): Promise<T[]> {
+    const elementToTarget: Map<T, DenylistTarget> = new Map(elements.map((element) => [element, targetBuild(element)]))
+    const areDenylisted = await this.denylist.areTargetsDenylisted(denylistRepo, Array.from(elementToTarget.values()))
+    return Array.from(elementToTarget.entries())
+      .filter(([, target]) => !isTargetDenylisted(target, areDenylisted))
+      .map(([element]) => element)
+  }
+
+  /** When an entity is denylisted, we don't want to show its content and metadata  */
```

Comment:
The entities that are used as param here have already been filtered, so is this method to filter the denylisted nested objects on those?

Reply:
As I said before, entities are not filtered when denylisted. We just hide some of their properties. That "hiding" is done here